Sources/BraintreeDataCollector/BTDataCollector.m

```diff
@@ -56,16 +61,19 @@ - (void)setUpKountWithDebugOn:(BOOL)debugLogging {
     self.kount = [KDataCollector sharedCollector];
     self.kount.debug = debugLogging;
 
-    CLAuthorizationStatus locationStatus = kCLAuthorizationStatusNotDetermined;
+    if ((locationStatus != kCLAuthorizationStatusAuthorizedWhenInUse && locationStatus != kCLAuthorizationStatusAuthorizedAlways) || ![CLLocationManager locationServicesEnabled]) {
```

Comment:
🤔 How do we know that `locationManagerDidChangeAuthorization` will be called to set the `locationStatus` before this method (`setUpKountWithDebugOn`) is called?

Reply:
From the docs for [`locationManagerDidChangeAuthorization`](https://developer.apple.com/documentation/corelocation/cllocationmanagerdelegate/3563956-locationmanagerdidchangeauthoriz) it states:
```
Core Location always calls locationManagerDidChangeAuthorization(_:) when the user’s action results in an authorization status change, and when your app creates an instance of [CLLocationManager](https://developer.apple.com/documentation/corelocation/cllocationmanager), whether your app runs in the foreground or in the background.
```
That said, doing some testing with adding breakpoints this seems somewhat inconsistent on initial load, so we may need to revisit. 🤔